recycle-bin.c

```diff
@@ -19,22 +19,25 @@ int wmain(int argc, wchar_t **argv) {
 		return 1;
 	}
 
-	size_t len = argc + 1;
+	size_t len = argc;
 
 	for (int i = 1; i < argc; i++) {
 		len += wcslen(argv[i]);
 	}
 
-	wchar_t *from = (wchar_t*)malloc(len * sizeof(wchar_t));
+	wchar_t *from = malloc(len * sizeof(wchar_t));
 
 	int pos = 0;
 
 	for (int i = 1; i < argc; i++) {
 		wcscpy(&from[pos], argv[i]);
-		pos += wcslen(argv[i]) + 1;
+		pos += wcslen(argv[i]);
+		from[pos] = '\0';
+		++pos;
 	}
 
-	from[len - 1] = '\0';
+	from[pos] = '\0';
+	// assert ++pos == len
```

Comment:
This just documents a post-requirement; `++pos` should be the same as `len` here.